runtime/parachains/src/paras_inherent.rs

```diff
@@ -1061,24 +1053,91 @@ mod tests {
 				assert_eq!(<scheduler::Pallet<Test>>::scheduled(), vec![]);
 
 				// Nothing is filtered out (including the backed candidates.)
-				let limit_para_inherent_data = Pallet::<Test>::create_inherent_inner(&inherent_data.clone()).unwrap();
-				assert!(limit_para_inherent_data != expected_para_inherent_data);
+				let multi_dispute_inherent_data = Pallet::<Test>::create_inherent_inner(&inherent_data.clone()).unwrap();
+				assert!(multi_dispute_inherent_data != expected_para_inherent_data);
+
+				assert_eq!(multi_dispute_inherent_data.disputes.len(), 2);
 
 				assert_eq!(
-					limit_para_inherent_data.disputes.len(),
-					2usize,
+					&multi_dispute_inherent_data.disputes[..2],
+					&expected_para_inherent_data.disputes[..2],
 				);
 
-				assert_eq!(
-					expected_para_inherent_data.disputes.len(),
-					3usize,
+				// The schedule is still empty prior to calling `enter`. (`create_inherent_inner` should not
+				// alter storage, but just double checking for sanity).
+				assert_eq!(<scheduler::Pallet<Test>>::scheduled(), vec![]);
+
+				assert_eq!(Pallet::<Test>::on_chain_votes(), None);
+				// Call enter with our 2 backed candidates
+				assert_ok!(Pallet::<Test>::enter(
+					frame_system::RawOrigin::None.into(),
+					multi_dispute_inherent_data,
+				));
+
+				assert_err!(
```

Comment:
Maybe a comment that testing `TooManyInclusionInherents` is not directly related to the point of this unit test, but just included here because its easy to add

Reply:
Well, it ensures that the prior data would have led to an overlength block, which ensures that we should indeed have filtered.